src/site/content/en/blog/catalina-system-ui/index.md

```diff
@@ -0,0 +1,246 @@
+
+```css
+h1 {
+  font-variation-settings: 'opsz' 20;
+}
+```
+
+Unfortunately, the default `opsz` value in the new Catalina font is `20`, and Chromium engineers were not prepared to apply `opsz` to the system font. This led to smaller sizes displaying too narrow. 
+
+To fix that, Chromium needed to apply `opsz` correctly to the system font. This lead to [Issue #1005969](https://crbug.com/1005969) getting fixed. Victory!…?
+
+### Not done yet
+
+This is where it got tricky: Chromium applied `opsz` but something did not look right still. System fonts on Mac have an additional font table called [`trak`](https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6trak.html), which tweaks horizontal spacing. While working on the fix, we noticed that on macOS, when retrieving horizontal metrics from a `CTFontRef CoreText` object, the `trak` metrics were already getting factored into the metrics results. Chromium's shaping library [HarfBuzz](https://github.com/harfbuzz/harfbuzz) needs metrics where the `trak` values are not yet factored in.
+
+<figure class="w-figure">
+  <img src="./weight-loss.png" alt="A master display of system-ui and all of it's font weight and variations in a list. Half of them have no weight differences applied.">
+</figure>
+
+Internally, [Skia](https://skia.org/) uses both the `CGFontRef` class from [CoreGraphics](https://developer.apple.com/documentation/coregraphics) and the `CTFontRef` class from [CoreText](https://developer.apple.com/documentation/coretext). Due to required internal conversions between those objects (used for keeping backwards compatibility and accessing needed APIs on both classes), Skia would lose weight information in certain circumstances and bold fonts would stop working. This was tracked in [Issue #1057654](https://crbug.com/1057654). 
+
+Skia still needs to support macOS 10.11 because Chromium still supports it. On 10.11 the San Francisco Text and San Francisco Display fonts weren't even variable fonts, but each was a family of separate fonts for every weight available. At some point their glyph IDs became out of sync with each other. So if Skia did text shaping (converting text into glyphs that can be drawn) with San Francisco Text, it would be gibberish if drawn with San Francisco Display, and vice versa. And even if Skia just asked for a different size macOS might switch to the other. It should be possible to always use one of the fonts and just scale it (using a matrix to scale it up instead of asking for a larger size) but CoreText has an issue where it will not scale sbix (color emoji) glyphs up (only down). It's a bit more complex than that. CoreText actually seems to cap the vertical extent after matrix application, which seems to be related to it not being able to draw emoji at 45 degree angles. In any event, if you want your emoji to be shown big, you need to make a copy of the font to get a big version.
```

Comment:
```suggestion
Skia still needs to support macOS 10.11 because Chromium still supports it. On 10.11 the "San Francisco Text" and "San Francisco Display" fonts weren't even variable fonts. Rather, each was a family of separate fonts for every weight available. At some point their glyph IDs became out of sync with each other. So if Skia did text shaping (converting text into glyphs that can be drawn) with "San Francisco Text", it would be gibberish if drawn with "San Francisco Display", and vice versa. And even if Skia just asked for a different size macOS might switch to the other. It should be possible to always use one of the fonts and just scale it (using a matrix to scale it up instead of asking for a larger size) but `CoreText` has an issue where it will not scale sbix (color emoji) glyphs up (only down). It's a bit more complex than that. `CoreText` actually seems to cap the vertical extent after matrix application, which seems to be related to it not being able to draw emoji at 45 degree angles. In any event, if you want your emoji to be shown big, you need to make a copy of the font to get a big version.
```